src/sessions/softphone-session-handler.ts

```diff
@@ -22,22 +95,321 @@ export default class SoftphoneSessionHandler extends BaseSessionHandler {
+
+    if (!participantsForUser.length) {
+      this.log('warn', 'user not found on conversation as a participant', { conversationId: update.id });
+      return;
+    }
+
+    /* one participant */
+    if (participantsForUser.length === 1) {
+      participant = participantsForUser[0];
+    }
+
+    /* find user participant with desired call state */
+    if (!participant && state) {
+      participant = participantsForUser.filter(p => p.calls.find(c => c.state === state))[0];
+    }
+
+    /* find user participant with a call */
+    if (!participant) {
+      participant = participantsForUser.find(p => p.calls.length)
+    }
+
+    /* find the most recent user participant */
+    if (!participant) {
+      participant = participantsForUser[0];
+    }
+
+    return participant;
+  }
+
+  getCallStateFromParticipant (participant: IConversationParticipantFromEvent): ICallStateFromParticipant | undefined {
+    const calls = participant?.calls;
+    const callLen = calls?.length;
+
+    if (!callLen) {
+      this.log('debug', 'no call found on participant', { userId: participant?.userId, participantId: participant?.id });
+      return;
+    }
+
+    /* if we only have one, use it */
+    if (callLen === 1) {
+      return calls[0];
+    }
+
+    const nonEndedCalls = calls.filter(c => !this.isEndedState(c));
+
+    /* if we only have one non-ended call, use it */
+    if (nonEndedCalls.length === 1) {
+      return nonEndedCalls[0];
+    }
+
+    /* else, grab the last one (which should be the most recent) */
+    return calls[callLen - 1];
+  }
+
+  private setCurrentSession (session: IExtendedMediaSession): void {
+    const loggy = () => ({
+      persistentConnectionSessionId: session.id,
+      currentConversationId: session.conversationId,
+      concurrentSoftphoneSessionsEnabled: this.sdk.concurrentSoftphoneSessionsEnabled(),
+      isPersistentConnectionEnabled: this.sdk.isPersistentConnectionEnabled(),
+      otherActiveSession: this.sdk.sessionManager.getAllActiveSessions()
+        .map(s => ({ sessionId: session.id, conversationId: session.conversationId }))
+    });
+    this.log('info', 'setting current session', loggy());
+
+    this.activeSession = session;
+    session.once('terminated', () => {
+      this.log('info', 'current session has terminated', loggy());
+      const otherActiveSessions = this.sdk.sessionManager.getAllActiveSessions();
+      if (this.sdk.isPersistentConnectionEnabled() && !!otherActiveSessions.length) {
+        this.log('debug', 'active session ended. Using next available session as persistent connection', {
+          terminatedSession: { sessionId: session.id, conversationId: session.conversationId },
+          selectedSession: { sessionId: otherActiveSessions[0].id, conversationId: otherActiveSessions[0].conversationId }
+        });
+        this.setCurrentSession(otherActiveSessions[0]);
+      } else {
+        this.activeSession = null;
+        (window as any).pc = null; // delete me
```

Comment:
Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/MyPureCloud/genesys-cloud-webrtc-sdk/pullRequest?prid=8384644)
```suggestion
        (window as unknown).pc = null; // delete me
```